packages/instant-apps-components/src/components/instant-apps-landing-page/instant-apps-landing-page.test.ts

```diff
@@ -1,88 +1,196 @@
 };
+const { enableSignIn, titleText, subtitleText, descriptionText, entryButtonText, iconImageScale, iconImage } = componentConfig;
+const CSS = {
+  BASE: 'instant-apps-landing-page',
+  titleText: 'instant-apps-landing-page__title-text',
+  subtitleText: 'instant-apps-landing-page__subtitle-text',
+  descriptionText: 'instant-apps-landing-page__description-text',
+  closed: 'instant-apps-landing-page--closed',
+  closedNoTransition: 'instant-apps-landing-page--closed-no-transition',
+  iconImage: 'instant-apps-landing-page__icon-image',
+  removeTransition: 'instant-apps-landing-page__remove-transition',
+  removePadding: 'instant-apps-landing-page__remove-padding',
+  alignment: 'instant-apps-landing-page__alignment--',
+  entryButton: 'instant-apps-landing-page__entry-button',
+  contentContainer: 'instant-apps-landing-page__content-container',
+  buttonContainer: 'instant-apps-landing-page__button-container',
+  signIn: 'instant-apps-landing-page__sign-in',
+  signInOverlay: 'instant-apps-landing-page__sign-in--overlay',
+  iconImageScale: {
+    s: ' instant-apps-landing-page__icon-image-scale--s',
+    m: ' instant-apps-landing-page__icon-image-scale--m',
+    l: ' instant-apps-landing-page__icon-image-scale--l',
+  },
+};
 
 describe('Landing Page', async () => {
-  const landingPage = document.createElement('instant-apps-landing-page');
-  landingPage.setAttribute('title-text', componentConfig.titleText);
-  landingPage.setAttribute('subtitle-text', componentConfig.subtitleText);
-  landingPage.setAttribute('description-text', componentConfig.descriptionText);
-  landingPage.setAttribute('entry-button-text', componentConfig.entryButtonText);
-  landingPage.setAttribute('icon-image-scale', componentConfig.iconImageScale);
-  landingPage.setAttribute('icon-image', componentConfig.iconImage);
-  document.body.appendChild(landingPage);
-  await new Promise(resolve => requestIdleCallback(resolve));
-  test('customization', async () => {
-    const element = document.querySelector('instant-apps-landing-page');
-    expect(element).toBeTruthy();
-    expect(element?.subtitleText).toBe('Hi this is only for testing purposes');
-    expect(element?.titleText).toBe('Vitest Instant App');
-    expect(element?.descriptionText).toBe("Áádóó naashnishgo, náásht'įįh łeets'óózh jiní. Diné be'elyá bee iináanii át'éego, shił yá'");
-    expect(element?.entryButtonText).toBe('Click me!');
+  type cases = [boolean, () => HTMLInstantAppsLandingPageElement];
+  type allTests = cases[];
 
-    expect(element?.iconImageScale).toBe('s');
-    expect(element?.iconImage).toBe('https://www.esri.com/content/dam/esrisites/en-us/arcgis/products/arcgis-instant-apps/assets/instant-apps-banner-fg.png');
-  });
+  const testCases: allTests = [
+    [
+      false,
+      function createNoSignIn() {
+        let ss = document.createElement('instant-apps-landing-page');
+        return ss;
+      },
+    ],
+    [
+      true,
+      function createSignIn() {
```

Comment:
`createNoSignIn` and `createSignIn` looks like it can be one function with an arg that determines whether to enable/disable sign in.